### PR TITLE
Add normalization the URI before cdniuc validation in URI Signing

### PR DESCRIPTION
**Pull request by dsouza93**
According to the latest URI Signing internet draft, the URI must be normalized before being checked against whatever is passed in the cdniuc container. 

The URI should be normalized according to both of the following RFCs
https://tools.ietf.org/html/rfc7230 - Section 2.7.3
https://tools.ietf.org/html/rfc3986 - Sections 6.2.2 and 6.2.3

### CODE REVIEW

**Comment by shinrich** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by jrushford** [comment]
[approve ci]

**Comment by jrushford** [comment]
[approve ci]

File: plugins/experimental/uri_signing/normalize.c [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+#include "common.h"
+#include <string.h>
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+/* Remove Dot Algorithm outlined in RFC3986 section 5.2.4
+ * Function writes normalizes path and writes to ret_buffer */
+int
+remove_dot_segments(const char *path, int path_ct, char *ret_buffer, int buff_ct)
```
Comment (jrushford) [inline plugins/experimental/uri_signing/normalize.c]: @dsouza93 I get the following warning when i build, try adding '#include "ts/ts.h" on line 30 after '#else' in common.h:

In file included from experimental/uri_signing/normalize.c:20:0:
experimental/uri_signing/normalize.c: In function 'remove_dot_segments':
experimental/uri_signing/common.h:31:26: warning: implicit declaration of function 'TSDebug' [-Wimplicit-function-declaration]
 #define PluginDebug(...) TSDebug("uri_signing", PLUGIN_NAME " " __VA_ARGS__)
                          ^
experimental/uri_signing/normalize.c:33:5: note: in expansion of macro 'PluginDebug'
     PluginDebug("Path buffer not large enough");
     ^~~~~~~~~~~
experimental/uri_signing/normalize.c: At top level:
cc1: warning: unrecognized command line option '-Wno-cast-function-type'

Comment (dsouza93) [reply plugins/experimental/uri_signing/normalize.c]: Should be resolved now

Comment (jrushford) [reply plugins/experimental/uri_signing/normalize.c]: looks good

**Comment by jrushford** [comment]
[approve ci]

**Pull request merged**